store/modules/wallet.js

```diff
@@ -0,0 +1,228 @@
+import { LIKECOIN_WALLET_CONNECTOR_CONFIG } from '@/constant/network';
+
+import {
+  WALLET_SET_ADDRESS,
+  WALLET_SET_SIGNER,
+  WALLET_SET_CONNECTOR,
+  WALLET_SET_METHOD_TYPE,
+  USER_SET_USER_INFO,
+  WALLET_SET_IS_LOGGING_IN,
+  WALLET_SET_IS_CONNECTING_WALLET,
+  AUTHCORE_SET_ACCESS_TOKEN,
+} from '../mutation-types';
+
+let likecoinWalletLib = null;
+
+const state = () => ({
+  address: '',
+  isConnectingWallet: false,
+  signer: null,
+  connector: null,
+  likerInfo: null,
+  isInited: null,
+  methodType: null,
+  loginAddress: '',
+  isLoggingIn: false,
+});
+
+const mutations = {
+  [WALLET_SET_ADDRESS](state, address) {
+    state.address = address;
+  },
+  [WALLET_SET_SIGNER](state, signer) {
+    state.signer = signer;
+  },
+  [WALLET_SET_IS_CONNECTING_WALLET](state, isConnecting) {
+    state.isConnectingWallet = isConnecting;
+  },
+  [WALLET_SET_IS_LOGGING_IN](state, isLoggingIn) {
+    state.isLoggingIn = isLoggingIn;
+  },
+  [USER_SET_USER_INFO](state, userInfo) {
+    state.likerInfo = userInfo;
+  },
+  [WALLET_SET_METHOD_TYPE](state, method) {
+    state.methodType = method;
+  },
+  [WALLET_SET_CONNECTOR](state, connector) {
+    state.connector = connector;
+  },
+};
+
+const getters = {
+  getAddress: state => state.address,
+  getSigner: state => state.signer,
+  loginAddress: state => state.loginAddress,
+  walletHasLoggedIn: state => !!state.address && !!state.loginAddress,
+  walletLoginMethod: state => state.methodType,
+  // eslint-disable-next-line max-len
+  walletIsMatchedSession: (state, getters) => getters.walletHasLoggedIn && state.address === state.loginAddress,
+  getConnector: state => state.connector,
+  getLikerInfo: state => state.likerInfo,
+  walletEmail: state => state.email,
+  walletEmailUnverified: state => state.emailUnverified,
+  walletHasVerifiedEmail: state => !!state.email,
+  walletIsLoggingIn: state => state.isConnectingWallet || state.isLoggingIn,
+};
+
+const actions = {
+  async getLikeCoinWalletLib() {
+    if (!likecoinWalletLib) {
+      likecoinWalletLib = await import(/* webpackChunkName: "likecoin_wallet" */ '@likecoin/wallet-connector');
+    }
+    return likecoinWalletLib;
+  },
+
+  async getConnector({ state, commit, dispatch }) {
+    if (state.connector) {
+      return state.connector;
+    }
+    const lib = await dispatch('getLikeCoinWalletLib');
+    const connector = new lib.LikeCoinWalletConnector({
+      ...LIKECOIN_WALLET_CONNECTOR_CONFIG,
+    });
+    commit(WALLET_SET_CONNECTOR, connector);
+    return connector;
+  },
+
+  async initWallet({
+    commit, dispatch,
+  }, {
+    method, accounts, offlineSigner,
+  }) {
+    if (!accounts[0]) return false;
+    commit(WALLET_SET_IS_CONNECTING_WALLET, true);
+    const connector = await dispatch('getConnector');
+    // Listen once per account
+    connector.once('account_change', async (currentMethod) => {
+      const connection = await connector.init(currentMethod);
+      dispatch('walletLogout');
```

Comment:
where is `walletLogout` defined?

Reply:
[🐛 Fix walletLogout undefined](https://github.com/likecoin/like-co/pull/1740/commits/8b456fdbf626e53478d683ab0a8ed202190e70b1)